### PR TITLE
feat: support for YAML output

### PR DESCRIPTION
**Pull request by aayushmau5**
**Description**

Currently, this library only supports JSON output, this PR adds support for YAML output. 

I'm using the [`js-yaml`](https://www.npmjs.com/package/js-yaml) library in order to convert the diff object to YAML string.

In order to pass the `outputType` to the library, the user has to pass `outputType: "json" | "yaml"` in the `config` object.

```js
diff(firstDocument, secondDocument, {
	outputType: "yaml"
});
```

The default value for the output type is `json`(thus, not a breaking change).

**Related issue(s)**

See https://github.com/asyncapi/diff/issues/75

### CODE REVIEW

File: API.md
Comment (magicmatatjahu) [inline API.md:25]: You don't need to push that changes, because it will be done by our bot after merge. Please revert that file :)

File: src/asyncapidiff.ts
Comment (magicmatatjahu) [inline src/asyncapidiff.ts:14]: I wonder if we should go with direction to pass as second argument to constructor object of options, not only one option. Do you understand what I mean? For future we can add more options and with shape like:

```js
constructor(output: string, options: { outputType: OutputType }) {
```

we will be able to extend that option object with additional fields without breaking changes. WDYT? :)

**Review by magicmatatjahu — CHANGES_REQUESTED** [review]

File: src/asyncapidiff.ts
Comment (aayushmau5) [reply src/asyncapidiff.ts:14]: Oh, yeah. I hadn't thought of that. Love this approach, thanks for the suggestion :+1:

File: src/asyncapidiff.ts
Comment (magicmatatjahu) [inline src/asyncapidiff.ts:32]: You can make helper function for that redundant logic (also in other functions):

```
function toProperFormat<T>(data: T, outputType: OutputType) {
  if (this.outputType === 'yaml') {
    return convertToYAML(data);
  }
  return data;
}
```

**Review by magicmatatjahu — COMMENTED** [review]
One comment :)

File: src/asyncapidiff.ts
Comment (aayushmau5) [reply src/asyncapidiff.ts:32]: What would be the right place for this function? In another module inside `helpers` directory or inside the `AsyncAPIDiff` class itself? :thinking:

Comment (magicmatatjahu) [reply src/asyncapidiff.ts:32]: `helpers` will be good place :)

File: src/types.ts
Comment (magicmatatjahu) [inline src/types.ts:48]: And last suggestion 😅  `yaml` can be also named as `yml` so we should also support it. Having `toProperFormat` function you need to make only one addition check in `if` statement :)

**Review by magicmatatjahu — APPROVED** [review]
LGTM! 🚀

**Comment by aayushmau5** [comment]
Sorry, had to fix merge conflicts :sweat_smile:

**Comment by jonaslagoni** [comment]
@aayushmau5 just tag @magicmatatjahu or re-request review when you need another review 🙂

**Comment by magicmatatjahu** [comment]
@aayushmau5 Yeah, as Jonas wrote, it's better to re-quest review by clicking that `circle` button:

image

because then we are informed by Toast on slack.

**Review by magicmatatjahu — APPROVED** [review]
LGTM! :)

**Pull request merged**
